### ISSUE TITLE
Problem: /usr/share/archivematica/dashboard/components/access/views.py:443 expects object to have resource

### ISSUE THREAD

**Issue by rvanlieshout** [archivematica/Issues#1265]
I've been playing around with Appraisal to transfer items from the backlog to Archivesspace. Found a record in ArchiveSpace and added a digital object. I currently have around 10 created digital objects which I'm unable to remove (and that might cause this issue?)

When I click Finalize arrangement I see an error in the Dashboard log pointing me to /usr/share/archivematica/dashboard/components/access/views.py:443 

I've checked the result of archival_object at that line and it's (converted to JSON for readability):

```json
{
    "lock_version": 23,
    "external_documents": [],
    "extents": [
        {
            "lock_version": 0,
            "system_mtime": "2020-05-15T08:16:51Z",
            "jsonmodel_type": "extent",
            "user_mtime": "2020-05-15T08:16:51Z",
            "number": "1",
            "last_modified_by": "ardv_rdevries",
            "portion": "whole",
            "create_time": "2020-05-15T08:16:51Z",
            "created_by": "ardv_rdevries",
            "extent_type": "meter"
        }
    ],
    "jsonmodel_type": "resource",
    "instances": [],
    "create_time": "2020-03-17T09:35:31Z",
    "publish": true,
    "finding_aid_script": "Latn",
    "title": "R.K. Parochie H. Isidorus te Mierlo",
    "related_accessions": [],
    "created_by": "ardv_rdevries",
    "subjects": [],
    "lang_materials": [
        {
            "lock_version": 0,
            "system_mtime": "2020-05-15T08:16:51Z",
            "jsonmodel_type": "lang_material",
            "notes": [],
            "user_mtime": "2020-05-15T08:16:51Z",
            "last_modified_by": "ardv_rdevries",
            "create_time": "2020-05-15T08:16:51Z",
            "created_by": "ardv_rdevries",
            "language_and_script": {
                "lock_version": 0,
                "system_mtime": "2020-05-15T08:16:51Z",
                "jsonmodel_type": "language_and_script",
                "language": "dut",
                "script": "Latn",
                "user_mtime": "2020-05-15T08:16:51Z",
                "last_modified_by": "ardv_rdevries",
                "create_time": "2020-05-15T08:16:51Z",
                "created_by": "ardv_rdevries"
            }
        }
    ],
    "deaccessions": [],
    "system_mtime": "2020-07-14T18:15:01Z",
    "linked_agents": [],
    "repository": {
        "ref": "/repositories/2"
    },
    "user_mtime": "2020-05-15T08:16:51Z",
    "rights_statements": [],
    "revision_statements": [],
    "linked_events": [],
    "classifications": [],
    "external_ids": [],
    "suppressed": false,
    "restrictions": false,
    "dates": [
        {
            "lock_version": 0,
            "system_mtime": "2020-05-15T08:16:51Z",
            "begin": "1995",
            "end": "2001",
            "jsonmodel_type": "date",
            "date_type": "inclusive",
            "user_mtime": "2020-05-15T08:16:51Z",
            "last_modified_by": "ardv_rdevries",
            "label": "creation",
            "create_time": "2020-05-15T08:16:51Z",
            "created_by": "ardv_rdevries",
            "expression": "1950 - 2000 (2001)"
        }
    ],
    "is_slug_auto": false,
    "level": "fonds",
    "notes": [
        {
            "type": "bioghist",
            "persistent_id": "4c67b9d6ee3903ad9b0d8c0f81969002",
            "subnotes": [
                {
                    "content": "'Geschiedenis en Organisatie'\n\nLorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vestibulum velit nec faucibus auctor. Morbi vel lorem mattis, laoreet magna nec, ullamcorper odio. Mauris maximus ipsum ut tincidunt vestibulum. Curabitur aliquet, augue quis rutrum convallis, ipsum sem venenatis nisl, sit amet vestibulum nunc justo id arcu. Ut ex metus, ultricies eu est sit amet, porttitor pellentesque arcu. Ut scelerisque orci vitae nulla ultrices feugiat. Aliquam erat volutpat. Vivamus rutrum efficitur sollicitudin. Donec consequat orci ut arcu consequat, sed vehicula ipsum sagittis. Vivamus in odio sed urna lacinia dictum at id ex. Mauris laoreet ultricies lectus, vel tempus purus interdum quis. Nam vestibulum mi leo, vel finibus sapien tempus nec.\n\nNam a sapien vel nunc tempus vulputate sit amet tincidunt velit. Quisque egestas, felis quis consequat mattis, ipsum turpis efficitur justo, vel auctor lectus nisi non lorem. Etiam lobortis in sem a scelerisque. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Nunc sed molestie lectus, id viverra nibh. Maecenas urna metus, semper nec laoreet vel, iaculis id mauris. Nullam fermentum efficitur tellus non convallis. Quisque dapibus non felis ut ullamcorper. Mauris fermentum cursus sapien. Pellentesque sit amet velit maximus, iaculis orci tempor, ullamcorper purus.\n\nIn ultrices iaculis lacus in porttitor. Donec vel consectetur metus. Ut volutpat ac orci ac scelerisque. Fusce id pretium ligula. Etiam eget dui condimentum, sagittis mauris vel, condimentum tellus. Pellentesque finibus auctor diam, sed eleifend dolor faucibus ac. Proin interdum placerat nisl, eget tristique orci euismod eget. Vivamus velit orci, rhoncus et tempus malesuada, fermentum dignissim ante. Duis dictum neque eu turpis tempor pretium at non magna. Nam tincidunt sapien eu justo euismod efficitur. Aliquam vitae volutpat ante. Fusce et commodo ex, quis aliquam ante. Sed purus felis, rhoncus quis gravida eu, vestibulum et ligula. Nullam velit metus, euismod eu massa vitae, tincidunt congue erat.\n\nProin posuere nulla non vestibulum vestibulum. Aliquam quis sapien turpis. Cras imperdiet pulvinar elit, nec maximus risus consectetur sed. In et augue purus. Nam consequat tellus mi. Curabitur lobortis velit eu varius semper. Vivamus elementum maximus nisl, at consectetur eros posuere eget.",
                    "publish": true,
                    "jsonmodel_type": "note_text"
                }
            ],
            "jsonmodel_type": "note_multipart",
            "publish": true
        },
        {
            "type": "custodhist",
            "persistent_id": "56c8b15b70ac662a810c5c09945d2dc9",
            "subnotes": [
                {
                    "content": "'Archief en inventarisatie'\n\nLorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vestibulum velit nec faucibus auctor. Morbi vel lorem mattis, laoreet magna nec, ullamcorper odio. Mauris maximus ipsum ut tincidunt vestibulum. Curabitur aliquet, augue quis rutrum convallis, ipsum sem venenatis nisl, sit amet vestibulum nunc justo id arcu. Ut ex metus, ultricies eu est sit amet, porttitor pellentesque arcu. Ut scelerisque orci vitae nulla ultrices feugiat. Aliquam erat volutpat. Vivamus rutrum efficitur sollicitudin. Donec consequat orci ut arcu consequat, sed vehicula ipsum sagittis. Vivamus in odio sed urna lacinia dictum at id ex. Mauris laoreet ultricies lectus, vel tempus purus interdum quis. Nam vestibulum mi leo, vel finibus sapien tempus nec.\n\nNam a sapien vel nunc tempus vulputate sit amet tincidunt velit. Quisque egestas, felis quis consequat mattis, ipsum turpis efficitur justo, vel auctor lectus nisi non lorem. Etiam lobortis in sem a scelerisque. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Nunc sed molestie lectus, id viverra nibh. Maecenas urna metus, semper nec laoreet vel, iaculis id mauris. Nullam fermentum efficitur tellus non convallis. Quisque dapibus non felis ut ullamcorper. Mauris fermentum cursus sapien. Pellentesque sit amet velit maximus, iaculis orci tempor, ullamcorper purus.\n\nIn ultrices iaculis lacus in porttitor. Donec vel consectetur metus. Ut volutpat ac orci ac scelerisque. Fusce id pretium ligula. Etiam eget dui condimentum, sagittis mauris vel, condimentum tellus. Pellentesque finibus auctor diam, sed eleifend dolor faucibus ac. Proin interdum placerat nisl, eget tristique orci euismod eget. Vivamus velit orci, rhoncus et tempus malesuada, fermentum dignissim ante. Duis dictum neque eu turpis tempor pretium at non magna. Nam tincidunt sapien eu justo euismod efficitur. Aliquam vitae volutpat ante. Fusce et commodo ex, quis aliquam ante. Sed purus felis, rhoncus quis gravida eu, vestibulum et ligula. Nullam velit metus, euismod eu massa vitae, tincidunt congue erat.\n\nProin posuere nulla non vestibulum vestibulum. Aliquam quis sapien turpis. Cras imperdiet pulvinar elit, nec maximus risus consectetur sed. In et augue purus. Nam consequat tellus mi. Curabitur lobortis velit eu varius semper. Vivamus elementum maximus nisl, at consectetur eros posuere eget.",
                    "publish": true,
                    "jsonmodel_type": "note_text"
                }
            ],
            "jsonmodel_type": "note_multipart",
            "publish": true
        },
        {
            "type": "odd",
            "persistent_id": "a20378e39ce6b1e03691e78987e3dae9",
            "subnotes": [
                {
                    "content": "Archiefbureau Ruerd de Vries\nRuerd de Vries\nKollum, 15 mei 2020",
                    "publish": true,
                    "jsonmodel_type": "note_text"
                }
            ],
            "jsonmodel_type": "note_multipart",
            "publish": true
        }
    ],
    "tree": {
        "ref": "/repositories/2/resources/559/tree"
    },
    "uri": "/repositories/2/resources/559",
    "last_modified_by": "ardv_rdevries",
    "id_0": "[RKM]",
    "finding_aid_language": "dut"
}
```

**Expected behaviour**

I would 1) expect myself  to be able to clean / remove the digital objects I've created here. When I click remove nothing happens and even if they disappear they are presented again when I search again. No log output there

2) be able to Finalize the arrangement

**Current behaviour**

1) Nothing happens.

2) Finalize arrangement results in a "Unable to start SIP; check dashboard logs."

It might be because I've placed / created something on the wrong place, but seem unable to resolve this issue using the interface.

**Steps to reproduce**

* Upload something to backlog
* Switch to Appraisal using panes Backlog and ArchivesSpace
* Create a digital object
* Drag file
* Finalize

**Your environment (version of Archivematica, operating system, other relevant details)**

* CentOS Linux release 7.8.2003 (Core)
* Installed version: 1.11.2-1

---

**For Artefactual use:**

Before you close this issue, you must check off the following:

- [ ] All pull requests related to this issue are properly linked
- [ ] All pull requests related to this issue have been merged
- [ ] A testing plan for this issue has been implemented and passed (testing plan information should be included in the issue body or comments)
- [ ] Documentation regarding this issue has been written and merged (if applicable)
- [ ] Details about this issue have been added to the release notes (if applicable)

**Comment by rvanlieshout** [comment]
ArchivesSpace version: 2.7.1 running on CentOS Linux release 8.2.2004 (Core)

**Comment by rvanlieshout** [comment]
I found out that direct call to ArchivesSpace also doesn't include a resource. I'm checking with ArchivesSpace to see if anything changes

**Comment by rvanlieshout** [comment]
Several requests are made to ArchiveSpace; some of which include a ["resource"]["href"] as expected, some of which don't.

**Comment by rvanlieshout** [comment]
I've done some debugging and the method `access_arrange_start_sip` in `dashboard/components/access/views.py:422` is called 3 times:

1.

```
archivematica.dashboard:views:access_arrange_start_sip:423:  access_arrange_start_sip <agentarchives.archivesspace.client.ArchivesSpaceClient object at 0x7fdaa3f2a9d0> <WSGIRequest: POST '/access/archivesspace/-repositories-2-archival_objects-35078/copy_from_arrange/'> arrange_path=/arrange/-repositories-2-archival_objects-35078f0f9b11f-a1a0-4a7a-a58f-94f13a3afeab, system=archivesspace, identifier=/repositories/2/archival_objects/35078
```

2.

```
archivematica.dashboard:views:access_arrange_start_sip:442:  archival object {u'lock_version': 6, u'external_documents': [], u'extents': [], u'jsonmodel_type': u'archival_object', u'ref_id': u'57bdccf1868a93401c5efeab19adcaf5', u'instances': [], u'create_time': u'2020-03-17T09:49:13Z', u'created_by': u'ardv_rdevries', u'ancestors': [{u'ref': u'/repositories/2/resources/559', u'level': u'fonds'}], u'title': u'Pastoor', u'display_string': u'Pastoor', u'publish': True, u'subjects': [], u'lang_materials': [], u'system_mtime': u'2020-07-16T09:30:42Z', u'linked_agents': [], u'repository': {u'ref': u'/repositories/2'}, u'has_unpublished_ancestor': False, u'user_mtime': u'2020-04-23T14:38:36Z', u'rights_statements': [], u'linked_events': [], u'resource': {u'ref': u'/repositories/2/resources/559'}, u'external_ids': [], u'suppressed': False, u'dates': [], u'component_id': u'1', u'is_slug_auto': False, u'level': u'series', u'notes': [], u'uri': u'/repositories/2/archival_objects/35078', u'last_modified_by': u'ardv_rdevries', u'restrictions_apply': False, u'position': 0}
```

3.

```
archivematica.dashboard:views:access_arrange_start_sip:445:  resource {u'lock_version': 23, u'external_documents': [], u'extents': [{u'lock_version': 0, u'system_mtime': u'2020-05-15T08:16:51Z', u'jsonmodel_type': u'extent', u'user_mtime': u'2020-05-15T08:16:51Z', u'number': u'1', u'last_modified_by': u'ardv_rdevries', u'portion': u'whole', u'create_time': u'2020-05-15T08:16:51Z', u'created_by': u'ardv_rdevries', u'extent_type': u'meter'}], u'jsonmodel_type': u'resource', u'instances': [], u'create_time': u'2020-03-17T09:35:31Z', u'publish': True, u'finding_aid_script': u'Latn', u'title': u'R.K. Parochie H. Isidorus te Mierlo', u'related_accessions': [], u'created_by': u'ardv_rdevries', u'subjects': [], u'lang_materials': [{u'lock_version': 0, u'system_mtime': u'2020-05-15T08:16:51Z', u'jsonmodel_type': u'lang_material', u'notes': [], u'user_mtime': u'2020-05-15T08:16:51Z', u'last_modified_by': u'ardv_rdevries', u'create_time': u'2020-05-15T08:16:51Z', u'created_by': u'ardv_rdevries', u'language_and_script': {u'lock_version': 0, u'system_mtime': u'2020-05-15T08:16:51Z', u'jsonmodel_type': u'language_and_script', u'language': u'dut', u'script': u'Latn', u'user_mtime': u'2020-05-15T08:16:51Z', u'last_modified_by': u'ardv_rdevries', u'create_time': u'2020-05-15T08:16:51Z', u'created_by': u'ardv_rdevries'}}], u'deaccessions': [], u'system_mtime': u'2020-07-24T12:28:24Z', u'linked_agents': [], u'repository': {u'ref': u'/repositories/2'}, u'user_mtime': u'2020-05-15T08:16:51Z', u'rights_statements': [], u'revision_statements': [], u'linked_events': [], u'classifications': [], u'external_ids': [], u'suppressed': False, u'restrictions': False, u'dates': [{u'lock_version': 0, u'system_mtime': u'2020-05-15T08:16:51Z', u'begin': u'1995', u'end': u'2001', u'jsonmodel_type': u'date', u'date_type': u'inclusive', u'user_mtime': u'2020-05-15T08:16:51Z', u'last_modified_by': u'ardv_rdevries', u'label': u'creation', u'create_time': u'2020-05-15T08:16:51Z', u'created_by': u'ardv_rdevries', u'expression': u'1950 - 2000 (2001)'}], u'is_slug_auto': False, u'level': u'fonds', u'notes': [{u'type': u'bioghist', u'persistent_id': u'4c67b9d6ee3903ad9b0d8c0f81969002', u'subnotes': [{u'content': u"'Geschiedenis en Organisatie'\n\nLorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vestibulum velit nec faucibus auctor. Morbi vel lorem mattis, laoreet magna nec, ullamcorper odio. Mauris maximus ipsum ut tincidunt vestibulum. Curabitur aliquet, augue quis rutrum convallis, ipsum sem venenatis nisl, sit amet vestibulum nunc justo id arcu. Ut ex metus, ultricies eu est sit amet, porttitor pellentesque arcu. Ut scelerisque orci vitae nulla ultrices feugiat. Aliquam erat volutpat. Vivamus rutrum efficitur sollicitudin. Donec consequat orci ut arcu consequat, sed vehicula ipsum sagittis. Vivamus in odio sed urna lacinia dictum at id ex. Mauris laoreet ultricies lectus, vel tempus purus interdum quis. Nam vestibulum mi leo, vel finibus sapien tempus nec.\n\nNam a sapien vel nunc tempus vulputate sit amet tincidunt velit. Quisque egestas, felis quis consequat mattis, ipsum turpis efficitur justo, vel auctor lectus nisi non lorem. Etiam lobortis in sem a scelerisque. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Nunc sed molestie lectus, id viverra nibh. Maecenas urna metus, semper nec laoreet vel, iaculis id mauris. Nullam fermentum efficitur tellus non convallis. Quisque dapibus non felis ut ullamcorper. Mauris fermentum cursus sapien. Pellentesque sit amet velit maximus, iaculis orci tempor, ullamcorper purus.\n\nIn ultrices iaculis lacus in porttitor. Donec vel consectetur metus. Ut volutpat ac orci ac scelerisque. Fusce id pretium ligula. Etiam eget dui condimentum, sagittis mauris vel, condimentum tellus. Pellentesque finibus auctor diam, sed eleifend dolor faucibus ac. Proin interdum placerat nisl, eget tristique orci euismod eget. Vivamus velit orci, rhoncus et tempus malesuada, fermentum dignissim ante. Duis dictum neque eu turpis tempor pretium at non magna. Nam tincidunt sapien eu justo euismod efficitur. Aliquam vitae volutpat ante. Fusce et commodo ex, quis aliquam ante. Sed purus felis, rhoncus quis gravida eu, vestibulum et ligula. Nullam velit metus, euismod eu massa vitae, tincidunt congue erat.\n\nProin posuere nulla non vestibulum vestibulum. Aliquam quis sapien turpis. Cras imperdiet pulvinar elit, nec maximus risus consectetur sed. In et augue purus. Nam consequat tellus mi. Curabitur lobortis velit eu varius semper. Vivamus elementum maximus nisl, at consectetur eros posuere eget.", u'publish': True, u'jsonmodel_type': u'note_text'}], u'jsonmodel_type': u'note_multipart', u'publish': True}, {u'type': u'custodhist', u'persistent_id': u'56c8b15b70ac662a810c5c09945d2dc9', u'subnotes': [{u'content': u"'Archief en inventarisatie'\n\nLorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vestibulum velit nec faucibus auctor. Morbi vel lorem mattis, laoreet magna nec, ullamcorper odio. Mauris maximus ipsum ut tincidunt vestibulum. Curabitur aliquet, augue quis rutrum convallis, ipsum sem venenatis nisl, sit amet vestibulum nunc justo id arcu. Ut ex metus, ultricies eu est sit amet, porttitor pellentesque arcu. Ut scelerisque orci vitae nulla ultrices feugiat. Aliquam erat volutpat. Vivamus rutrum efficitur sollicitudin. Donec consequat orci ut arcu consequat, sed vehicula ipsum sagittis. Vivamus in odio sed urna lacinia dictum at id ex. Mauris laoreet ultricies lectus, vel tempus purus interdum quis. Nam vestibulum mi leo, vel finibus sapien tempus nec.\n\nNam a sapien vel nunc tempus vulputate sit amet tincidunt velit. Quisque egestas, felis quis consequat mattis, ipsum turpis efficitur justo, vel auctor lectus nisi non lorem. Etiam lobortis in sem a scelerisque. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Nunc sed molestie lectus, id viverra nibh. Maecenas urna metus, semper nec laoreet vel, iaculis id mauris. Nullam fermentum efficitur tellus non convallis. Quisque dapibus non felis ut ullamcorper. Mauris fermentum cursus sapien. Pellentesque sit amet velit maximus, iaculis orci tempor, ullamcorper purus.\n\nIn ultrices iaculis lacus in porttitor. Donec vel consectetur metus. Ut volutpat ac orci ac scelerisque. Fusce id pretium ligula. Etiam eget dui condimentum, sagittis mauris vel, condimentum tellus. Pellentesque finibus auctor diam, sed eleifend dolor faucibus ac. Proin interdum placerat nisl, eget tristique orci euismod eget. Vivamus velit orci, rhoncus et tempus malesuada, fermentum dignissim ante. Duis dictum neque eu turpis tempor pretium at non magna. Nam tincidunt sapien eu justo euismod efficitur. Aliquam vitae volutpat ante. Fusce et commodo ex, quis aliquam ante. Sed purus felis, rhoncus quis gravida eu, vestibulum et ligula. Nullam velit metus, euismod eu massa vitae, tincidunt congue erat.\n\nProin posuere nulla non vestibulum vestibulum. Aliquam quis sapien turpis. Cras imperdiet pulvinar elit, nec maximus risus consectetur sed. In et augue purus. Nam consequat tellus mi. Curabitur lobortis velit eu varius semper. Vivamus elementum maximus nisl, at consectetur eros posuere eget.", u'publish': True, u'jsonmodel_type': u'note_text'}], u'jsonmodel_type': u'note_multipart', u'publish': True}, {u'type': u'odd', u'persistent_id': u'a20378e39ce6b1e03691e78987e3dae9', u'subnotes': [{u'content': u'Archiefbureau Ruerd de Vries\nRuerd de Vries\nKollum, 15 mei 2020', u'publish': True, u'jsonmodel_type': u'note_text'}], u'jsonmodel_type': u'note_multipart', u'publish': True}], u'tree': {u'ref': u'/repositories/2/resources/559/tree'}, u'uri': u'/repositories/2/resources/559', u'last_modified_by': u'ardv_rdevries', u'id_0': u'[RKM]', u'finding_aid_language': u'dut'}
```

It would make sense that 2 includes a ["resource"]["url"] and that 3 doesn't. Why is 3 triggered?

**Comment by rvanlieshout** [comment]
Could it be as simple as a arrangement that is still left and needs to be deleted? Can I see those arrangements somewhere?

**Comment by rvanlieshout** [comment]
I've deleted all my data from the Archivematica instance and tried again; without any luck.

Added a single digital file to an entry in ArchivesSpace on the Appraisal-tab, dragged a single file from the backlog and pressed finalize.

The code in /usr/share/archivematica/dashboard/components/access/views.py:422 def access_arrange_start_sip gets trigged twice:

1. mapping.identifier: /repositories/2/archival_objects/35591
2. mapping.identifier: blank

The second gets a resource on `archival_object = client.get_record(mapping.identifier)` which obviously is wrong as it expects this to be an archival_object and not a resource.

Anybody available to share some knowledge here?
